osbuild/testutil/net.py

```diff
@@ -24,6 +28,29 @@ def __init__(self, *args, **kwargs):  # pylint: disable=unused-argument
 
 from .atomic import AtomicCounter
 
+# tree prefix components
+space =  '    '
+branch = '│   '
+# pointers:
+tee =    '├── '
+last =   '└── '
+
+
+def tree(dir_path: Path, prefix: str=''):
+    """A recursive generator, given a directory Path object
+    will yield a visual tree structure line by line
+    with each line prefixed by the same characters
+    """
+    contents = list(dir_path.iterdir())
```

Comment:
A minor thing, since this is basically test code I don't hold a very strong opinion.

There's no need to turn this into a list, the `zip()` call below can consume an iterator directly though it will lead to a bit of restructuring down below since you can't easily pull the length anymore (you could use `izip_longest` instead).

Reply:
Not a Python expert, but there is a length required prior iterating, so in fact list is required perhaps?!? Frankly, I copy-pasted this code from SO :-)

This can be removed, tho, it saved me some time trying to figure out why I am getting 404s - the mock repository was not an archive type so the client was searching for compressed files.